code/__DEFINES/components.dm

```diff
@@ -213,11 +213,22 @@
 #define COMSIG_OBJ_SETANCHORED "obj_setanchored"				//called in /obj/structure/setAnchored(): (value)
 #define COMSIG_OBJ_DEFAULT_UNFASTEN_WRENCH "obj_default_unfasten_wrench"	//from base of code/game/machinery
 #define COMSIG_OBJ_UPDATE_ICON	"obj_update_icon"				//Called by obj to itself whenever update_icon is run. Useful for containers that want to detect this change
+#define COMSIG_OBJ_ATTACK_ANIMAL "obj_attack_animal"			//from base obj/attack_animal()
 
 // /obj/machinery signals
 #define COMSIG_MACHINERY_BROKEN "machinery_broken"				//from /obj/machinery/obj_break(damage_flag): (damage_flag)
 #define COMSIG_MACHINERY_POWER_LOST "machinery_power_lost"			//from base power_change() when power is lost
 #define COMSIG_MACHINERY_POWER_RESTORED "machinery_power_restored"	//from base power_change() when power is restored
+#define COMSIG_AIRLOCK_BUMPOPEN "airlock_bumpopen"					//from airlock bumpopen()
+	#define COMSIG_ELECTRIFIED_SHOCK 1
+#define COMSIG_AIRLOCK_ATTACKBY_CONDUCTIVE "airlock_attackby_conductive" //from airlock attackby()
+#define COMSIG_AIRLOCK_ATTACKBY_POWERCROWBAR "airlock_attackby_powercrowbar"
+#define COMSIG_AIRLOCK_ATTACK_ALIEN "airlock_attack_alien"
+#define COMSIG_ELECTRIFICATION_CHANGE "electrification_change"
+#define COMSIG_AIRLOCK_AI_ALTCLICK "airlock_ai_altclick"

```

Comment:
Signal names referencing their singe use are icky. Name this something like AIRLOCK_SILICON_ELECTRIFY

Reply:
You have this backwards, signals should be named after what's happening to cause it, not what the signal should do